presto-parquet/src/main/java/io/prestosql/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -255,9 +256,8 @@ public static Domain getDomain(
             // the special case where min == max and an incorrect ordering would not be material to the result. PARQUET-1026 made binary stats
             // available and valid in that special case
             if (binaryStatistics.genericGetMin().equals(binaryStatistics.genericGetMax())) {
-                long timestampValue = getTimestampMillis(binaryStatistics.genericGetMax());
-                timestampValue = timeZone.convertUTCToLocal(timestampValue) * MICROSECONDS_PER_MILLISECOND;
-                return createDomain(type, hasNullValue, new ParquetTimestampStatistics(timestampValue, timestampValue));
+                PrestoTimestampEncoder<?> prestoTimestampEncoder = createTimestampEncoder((TimestampType) type, timeZone);
+                return ParquetTimestampUtils.createDomain(prestoTimestampEncoder, binaryStatistics.genericGetMax(), hasNullValue);
```

Comment:
using `prestoTimestampEncoder` here is correct only because min=max
please add a comment